src/main/java/org/mifos/connector/ams/zeebe/ZeebeUtil.java

```diff
@@ -39,4 +57,66 @@ public static <T> T zeebeVariable(Exchange exchange, String name, Class<T> clazz
         }
         return (T) content;
     }
+
+    public static LoanRepaymentDTO setLoanRepaymentBody(Exchange exchange) {
+        LoanRepaymentDTO loanRepaymentDTO=new LoanRepaymentDTO();
+        loanRepaymentDTO.setDateFormat("dd MMMM yyyy");
+        loanRepaymentDTO.setLocale("en");
+        loanRepaymentDTO.setPaymentTypeId("1");
+        loanRepaymentDTO.setTransactionAmount(exchange.getProperty("amount").toString());
+        // Setting date
+        Format formatter = new SimpleDateFormat("dd MMMM yyyy");
+        String currDate = formatter.format(new Date());
+        loanRepaymentDTO.setTransactionDate(currDate);
+        return loanRepaymentDTO;
+    }
+
+    public  Exchange setAccountTypeandNumber(Exchange e,String accountNo){
+        String accountTypeIdentifier="";
+        String accountNumber="";
+        int n=accountNo.length();
+        // Separating account id and prefix
+        for(String s:accountPrefixes){
+            if(accountNo.startsWith(s)){
+                accountNumber=accountNo.substring(s.length(),n);
+                accountTypeIdentifier=accountNo.substring(0,s.length());
+                break;
+            }
+        }
+        logger.debug("Accout number:{}, Identifier:{}",accountNumber,accountTypeIdentifier);
+        e.setProperty(ACCOUNT_NUMBER,accountNumber);
+        e.setProperty(ACCOUNT_IDENTIFIER,accountTypeIdentifier);
+        return e;
+    }
+    public static String getValueofKey(List<CustomData> customData, String key){
+        for(CustomData obj:customData){
+            String keyString=obj.getKey();
+            if(keyString.equalsIgnoreCase(key)){
+                return obj.getValue().toString();
+            }
+        }
+        return null;
+    }
+    public static String convertGsmaTransfertoTransactionChannel(GsmaTransfer gsmaTransfer, Object property) throws JsonProcessingException {
+        TransactionChannelRequestDTO transactionChannelRequestDTO=new TransactionChannelRequestDTO();
+        String msisdn= gsmaTransfer.getPayer().get(0).getPartyIdIdentifier();
+        String accountId=property.toString();
+
+        String amount=gsmaTransfer.getAmount().trim();
+        List<CustomData> customData=gsmaTransfer.getCustomData();
+        String currency=getValueofKey(customData,"currency");
```

Comment:
As per the API design, we have the currency as a field, so no need to pass it and retrieve it from the  custom data.